src/import_whatsapp.js

```diff
@@ -252,13 +258,13 @@ const setImgMsgRegex = (fileType) => {
 		// console.info("ios format");
 		// iOS format
 		messageRegex =
-			/\[(\d{2}\/\d{2}\/\d{4}),\s(\d{1,2}:\d{2}:\d{2}\s(?:AM|PM))\]\s(.*?):\s(.+?)(?=\n\[|$)/gs;
+			/\[(\d{2,4}\/\d{2}\/\d{2,4}),\s(\d{1,2}:\d{2}:\d{2}\s(?:AM|PM))\]\s(.*?):\s(.+?)(?=\n\[|$)/gs;
 		imgFileRegex = /<attached: (\d+-[\w\-_]+\.(jpg|jpeg|png|gif))>/gim;
 	} else if (fileType.match(/\d{2}\//)) {
 		// console.info("android format");
 		// Android format
 		messageRegex =
-			/(\d{2}\/\d{2}\/\d{4}),?\s(\d{1,2}:\d{2})(?:\s?(?:AM|PM|am|pm))?\s-\s(.*?):[\t\f\cK ]((.|\n)*?)(?=(\n\d{2}\/\d{2}\/\d{4})|$)/g;
+			/(\d{2,4}\/\d{2}\/\d{2,4}),?\s(\d{1,2}:\d{2})(?:\s?(?:AM|PM|am|pm))?\s-\s(.*?):[\t\f\cK ]((.|\n)*?)(?=(\n\d{2}\/\d{2}\/\d{4})|$)/g;
```

Comment:
The day part doesn't need to be able to accept 4 chars, and you need to edit the lookahead to allow YY as well.
```suggestion
			/(\d{2}\/\d{2}\/\d{2,4}),?\s(\d{1,2}:\d{2})(?:\s?(?:AM|PM|am|pm))?\s-\s(.*?):[\t\f\cK ]((.|\n)*?)(?=(\n\d{2}\/\d{2}\/\d{2,4})|$)/g;
```

Reply:
can't it sometimes come in yyyy/mm/dd as well though? I thought year isn't always last, or was that just because of the format we create?

Reply:
If that's the case then formatDateString needs to be updated to figure out if day/year comes first. At the moment it assumes day (line 133)